src/transactions/TransactionFrame.cpp

```diff
@@ -1685,7 +1685,8 @@ TransactionFrame::applyExpirationBumps(Application& app, AbstractLedgerTxn& ltx)
         if (lte && isSorobanDataEntry(lte.current().data))
         {
             // Must enforce minimum expirations on write
-            bump(lte.current(), autoBumpEnabled(lte.current()), true);
+            bump(lte.current(),
+                 autoBumpEnabled(lte.current()) && autoBumpLedgers > 0, true);
```

Comment:
nit: swap the conditions to not touch lte when autobump is disabled (same below)?

Reply:
Done. I also updated the read section again because we would still `load` and update lastModified.